src/main/kotlin/me/zeroeightsix/kami/feature/module/AutoEat.kt

```diff
@@ -1,21 +1,44 @@
 package me.zeroeightsix.kami.feature.module
 
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
 import me.zero.alpine.listener.EventHandler
 import me.zero.alpine.listener.Listener
 import me.zeroeightsix.kami.event.TickEvent.InGame
 import me.zeroeightsix.kami.mixin.client.IKeyBinding
+import me.zeroeightsix.kami.setting.SettingVisibility
 import net.minecraft.client.options.KeyBinding
 import net.minecraft.item.ItemGroup
 import net.minecraft.item.ItemStack
 import net.minecraft.util.Hand
+import kotlin.math.roundToInt
 
 @Module.Info(name = "AutoEat", description = "Automatically eat when hungry", category = Module.Category.PLAYER)
 object AutoEat : Module() {
     private var eating: Hand? = null
     private var oldSlot: Int? = null
 
+    @Setting(comment = "Use the threshold value to activate AutoEat instead of the food hunger value")
+    private var useCustomThreshold = false
+
+    @Setting(comment = "If the amount of free hunger points goes above this, AutoEat activates")
+    @SettingVisibility.Method("shouldUseCustomThreshold")
+    // there is absolutely no reason for this to be a double but i can't use an Int
+    private var threshold: @Setting.Constrain.Range(min = 0.0, max = 20.0, step = 1.0) Int = 5
```

Comment:
The step of `1` is rather sensitive when using the GUI. It's an `Int`, so the setting won't have a floating point, but lowering the step *should* still make the sensitivity lower.

Reply:
true, but that seems like a pretty dirty solution, given that this could be fixed otherwise and there is no other advantage to using a double

Reply:
You don't need to make it a double. Lowering the step should only decrease sensitivity, but since it's an int, I'm pretty sure it should still have a step of 1.. I think.

I don't remember how I made this.

Reply:
if i set the step to a non-integer then it crashes once the settings are opened

Reply:
Ehh, doesn't matter enough for me to look into it.